### PR TITLE
Decrease dependency in `toolz` to optimize curried functions.

### PR DESCRIPTION
**Pull request by uriva**

### CODE REVIEW

File: gamla/functional.py [hunk 1]
```diff
@@ -500,3 +506,38 @@ def contains(y):
...
+        return y < x
+
+    return less_than
+
+
+def less_equals(x):
+    def less_equals(y):
+        return y <= x
+
+    return less_equals
```
Comment (erez-work) [inline gamla/functional.py:543]: Can you please add truediv and multiply?
(operator.truediv, operator.imul)

File: gamla/functional_test.py [hunk 2]
```diff
@@ -224,7 +234,7 @@ async def test_drop_last_while():
 def test_apply_spec():
     assert (
         functional_generic.apply_spec(
-            {"identity": toolz.identity, "increment": lambda x: x + 1},
+            {"identity": functional.identity, "increment": lambda x: x + 1},
```
Comment (erez-work) [inline gamla/functional_test.py:237]: Consider `add(1)`

File: gamla/tree_test.py [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
 
 def test_get_leaves_by_ancestor_predicate_no_matches():
-    fn = tree.get_leaves_by_ancestor_predicate(operator.eq("x"))
+    fn = tree.get_leaves_by_ancestor_predicate(functional.equals("x"))
     assert fn({"t": {"t": ("1")}}) == ()
 
 
 def test_filter_leaves():
-    fn = tree.filter_leaves(operator.lt(3))
+    fn = tree.filter_leaves(functional.greater_than(3))
```
Comment (erez-work) [inline gamla/tree_test.py:15]: I guess you want `less_than`

**Review by erez-work — APPROVED** [review]
Just mind the last comment about `lt`

File: gamla/functional_test.py (hunk 2 above)
Comment (uriva) [reply gamla/functional_test.py:237]: oo nice

**Pull request merged**